### ISSUE TITLE
Laravel 5.0.26 - Class 'IronMQ' not found

### ISSUE THREAD

**Issue by pfeiferchristopher** [laravel/framework#8263]
I'm having a terrible time with IronMQ and Laravel 5.0.26. Currently I am locked onto `"iron-io/iron_mq": "~1.5"` which is what the Laravel 5.0 documentation states is required. However, I am running into the follow errors as I've seen many times with other Laravel installations.
```php
FatalErrorException in IronConnector.php line 49:
Class 'IronMQ' not found
```
After going through IronConnector.php I see that it refers to `IronMQ` directly without any namespacing
```php
use IronMQ

...

$iron = new IronMQ($ironConfig);
```
but upon going through Composer's autoloader it seems as though IronMQ is now registered through `IronMQ\\IronMQ` namespace in `vendor/composer/autoload_classmap.php`
```php
'IronMQ\\IronMQ' => $vendorDir . '/iron-io/iron_mq/src/IronMQ.php',
```
I can't figure it out, I've tried different versions of IronMQ. First `2.*` and just `1.*` but either got the same error, or errors sending/receiving messages from Iron Queue. Although if I change `IronConnection.php` as follows I get no errors.
```php
use IronMQ\IronMQ;
```

**Comment by GrahamCampbell** [comment]
`use IronMQ\IronMQ;` is wrong. That's for 2.x.

**Comment by GrahamCampbell** [comment]
Hmmm. Is nobody even using ironmq then? It's been broken for ages.

**Comment by GrahamCampbell** [comment]
I can't repllicate. This works fine. The IronMQ class does exist: https://github.com/iron-io/iron_mq_php/blob/1.5.2/IronMQ.class.php#L140.

**Comment by pfeiferchristopher** [comment]
So then why is composer loading it via `IronMQ\IronMQ`? Should I require it without the `~`?

**Comment by GrahamCampbell** [comment]
It's not for me. Maybe your version of composer is outdated.

**Comment by onbjerg** [comment]
I am using IronMQ. No issue here.

**Comment by pfeiferchristopher** [comment]
So I have two environments staging, and production. Currently both are using the same exact `composer.json` and `composer.lock` but upon completely removing `vendor/` from both and running `composer install` and having everything finish successfully the staging environment still has IronMQ namespaced as `IronMQ\\IronMQ` and I can not find any reason for the differences... any help would be much appreciated. Thankfully Production is functional so I do have some time to fix this.

**Comment by pfeiferchristopher** [comment]
I figured out the issue. In IronMQ `1.5.3` they did indeed namespace it. So when I used that `~1.5` like the Laravel Docs instruct it would install `>=1.5` meaning `1.5.3` was being installed. I specificed `1.5.2` and now it's back to un-namespaced.

**Comment by GrahamCampbell** [comment]
That's not true. 1.5.3 is NOT namespaced. Look at the code: https://github.com/iron-io/iron_mq_php/blob/1.5.3/IronMQ.class.php.

**Comment by pfeiferchristopher** [comment]
But here it is https://github.com/iron-io/iron_mq_php/blob/v1/src/IronMQ.php would branch `v1` not be what `1.*` would load? I'm not very good at navigation Repo's and all.

**Comment by pfeiferchristopher** [comment]
Okay okay @GrahamCampbell I do apologize. As I stated I am horrible and unaccustomed to navigating Git repos but I do now see how to look at tags and such.

Upon looking at my composer.lock file and the composer.json within `iron-io/iron_mq` when I used `1.*` or `~1.5` I was getting version `2.0` installed. No clue why, but that is why it wasn't the correct version. So I locked onto `"iron-io/iron_mq": "1.5.3"` and I finally have the right versions and was able to re-implement my queue code and test it.

Again I do apologize for any confusion I have caused but I still blame composer and it was installing the wrong version.
